Apps/Database/src/Context/GatewayDbContext.cs

```diff
@@ -175,13 +175,26 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<UserPreference>()
                 .HasKey(c => new { c.HdId, c.Preference });
 
+            modelBuilder.Entity<Communication>()
+                .HasOne<CommunicationTypeCode>()
+                .WithMany()
+                .HasPrincipalKey(k => k.StatusCode)
+                .HasForeignKey(k => k.CommunicationTypeCode);
+
+            modelBuilder.Entity<Communication>()
+                .HasOne<CommunicationStatusCode>()
+                .WithMany()
+                .HasPrincipalKey(k => k.StatusCode)
```

Comment:
Is this right, we status code linked to two other columns or we have the same identifier in the code tables?

Reply:
I have created a new CommunicationStatusCode table for this CommunicationStatusCode. It is completely separated from the EmailStatusCode one(s).